app/scripts/modules/core/src/presentation/forms/validation/categories.spec.ts

```diff
@@ -1,29 +1,64 @@
-import { categorizeErrorMessage, categorizeErrors } from './categorizedErrors';
+import {
+  asyncMessage,
+  categorizeValidationMessage,
+  categorizeValidationMessages,
+  errorMessage,
+  infoMessage,
+  messageMessage,
+  successMessage,
+  warningMessage,
+} from './categories';
 
 describe('categorizeErrorMessage', () => {
   it('returns an array of length 2', () => {
-    const result = categorizeErrorMessage('');
+    const result = categorizeValidationMessage('');
     expect(Array.isArray(result)).toBeTruthy();
     expect(result.length).toBe(2);
   });
 
   it('returns the matching category key if the input starts with a category label and ": "', () => {
-    expect(categorizeErrorMessage('Error: ')[0]).toEqual('error');
-    expect(categorizeErrorMessage('Warning: ')[0]).toEqual('warning');
-    expect(categorizeErrorMessage('Async: ')[0]).toEqual('async');
-    expect(categorizeErrorMessage('Message: ')[0]).toEqual('message');
+    expect(categorizeValidationMessage('Error: ')[0]).toEqual('error');
+    expect(categorizeValidationMessage('Warning: ')[0]).toEqual('warning');
+    expect(categorizeValidationMessage('Async: ')[0]).toEqual('async');
+    expect(categorizeValidationMessage('Message: ')[0]).toEqual('message');
   });
 
   it('returns "error" category key by default if no category label is present', () => {
-    expect(categorizeErrorMessage('there was an error')[0]).toEqual('error');
+    expect(categorizeValidationMessage('there was an error')[0]).toEqual('error');
   });
 
   it('returns the entire error message if no category label is present', () => {
-    expect(categorizeErrorMessage('there was an error')[1]).toEqual('there was an error');
+    expect(categorizeValidationMessage('there was an error')[1]).toEqual('there was an error');
   });
 
   it('returns the error message without the label prefix', () => {
-    expect(categorizeErrorMessage('Warning: something sorta bad')[1]).toEqual('something sorta bad');
+    expect(categorizeValidationMessage('Warning: something sorta bad')[1]).toEqual('something sorta bad');
+  });
+});
+
+describe('category message builder', () => {
+  it('asyncMessage should prefix a message with Async:', () => {
+    expect(asyncMessage('the quick brown fox')).toBe('Async: the quick brown fox');
+  });
+
+  it('errorMessage should prefix a message with Error:', () => {
+    expect(errorMessage('the quick brown fox')).toBe('Error: the quick brown fox');
+  });
+
+  it('infoMessage should prefix a message with Info:', () => {
+    expect(infoMessage('the quick brown fox')).toBe('Info: the quick brown fox');
+  });
+
+  it('messageMessage should prefix a message with Message:', () => {
```

Comment:
not sure how i feel about `messageMessage`

Reply:
![giphy](https://user-images.githubusercontent.com/15936279/66216522-a25b8a80-e693-11e9-9259-2f504b3245ab.gif)